dali/operators/reader/nvdecoder/sequencedesc.h

```diff
@@ -92,22 +87,17 @@ struct SequenceWrapper {
     LOG_LINE << event_ << " synchronized!" << std::endl;
   }
 
-  void share_frames(TensorList<GPUBackend> &frames) {
-    void *current_sequence = sequence.raw_mutable_data();
-    auto shape = TensorListShape<>::make_uniform(count, frame_shape());
-
-    frames.ShareData(
-      current_sequence,
-      sequence.type().size() * count * height * width * channels,
-      shape,
-      sequence.type());
-  }
-
   TensorShape<3> frame_shape() const {
     return TensorShape<3>{height, width, channels};
   }
 
-  Tensor<GPUBackend> sequence;
+  TensorShape<4> shape() const {
+    return TensorShape<4>{count, height, width, channels};
+  }
+
+  // non-owning tensor that should only share data allocated somwhere else
```

Comment:
Is it really true from this sturcture's perspective? It's used like this, but would anything break if it was used differently?

Reply:
It should still work, from this class perspective.